harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -153,7 +154,14 @@ def _set_data_loaders(self) -> None:
                 )
             self.training_loader = train_data
 
-        self.context._epoch_len = len(self.training_loader)
+        # All workers use the chief's definition of epoch lengths, which is based on the training
+        # loader's len. If this len does not exist, epoch durations cannot be deduced, and they
+        # default to max int.
+        try:
+            epoch_len = len(self.training_loader)
+        except TypeError:
+            epoch_len = sys.maxsize
+        self.context._epoch_len = self.context.distributed.broadcast(epoch_len)
```

Comment:
this is a good start, but what do we do about variable-length datasets?

Eventually we'll need to break the assumption that the dataset length doesn't change, and we'll have to keep counters of epoch idx, rather than calculate it with modulo math.

... why not now?